api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -53,61 +70,70 @@
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.ScopedProxyMode;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.annotation.DirtiesContext.ClassMode;
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.context.annotation.RequestScope;
 
 @RunWith(SpringRunner.class)
 @DataJpaTest
 @Import(LiquibaseAutoConfiguration.class)
 @AutoConfigureTestDatabase(replace= AutoConfigureTestDatabase.Replace.NONE)
 @DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
-@Transactional(propagation = Propagation.NOT_SUPPORTED)
+@Transactional(propagation = Propagation.SUPPORTS)
 public class WorkspacesControllerTest {
   private static final Instant NOW = Instant.now();
   private static final long NOW_TIME = Timestamp.from(NOW).getTime();
   private static final FakeClock CLOCK = new FakeClock(NOW, ZoneId.systemDefault());
+  private static final String LOGGED_IN_USER_EMAIL = "bob@gmail.com";
 
   @TestConfiguration
-  @Import(WorkspaceServiceImpl.class)
-  @MockBean(FireCloudService.class)
+  @Import({
+    WorkspacesController.class,
+    WorkspaceServiceImpl.class
+  })
+  @MockBean(value={FireCloudService.class, CohortService.class})
   static class Configuration {
     @Bean
     Clock clock() {
       return CLOCK;
     }
+
+    @Bean
```

Comment:
Is this supposed to be here?

If you return user here does it obviate the need for setting user provider?

Reply:
Added a comment. This is needed for the initial wireup; actual usage is stubbed out with `setUserProvider()` below.